typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -1,81 +1,181 @@
+        StaticAggregationHookFactory__factory.connect(
+          args.addresses.staticAggregationHookFactory,
+          signer,
+        ),
+      domainRoutingIsmFactory: DomainRoutingIsmFactory__factory.connect(
+        args.addresses.domainRoutingIsmFactory,
+        signer,
+      ),
+    };
   }
 
   public async read(): Promise<IsmConfig> {
     return await this.reader.deriveIsmConfig(this.args.addresses.deployedIsm);
   }
 
   public async update(config: IsmConfig): Promise<EthersV5Transaction[]> {
-    throw new Error('Method not implemented.');
-
     const destination = this.multiProvider.getChainName(this.args.chain);
-    await this.creator.update({
-      destination,
-      config,
-      existingIsmAddress: this.args.addresses.deployedIsm,
-    });
+
+    if (typeof config === 'string') {
+      this.logger.debug('Skipping update for config of type Address.');
+      return [];
+    }
+
+    const ismType = config.type;
+    const logger = this.logger.child({ destination, ismType });
+
+    logger.debug(
+      `Updating ${ismType} on ${destination} ${
+        origin ? `(for verifying ${origin})` : ''
+      }`,
+    );
+
+    let contract: DeployedIsm;
+    if (ismType === IsmType.ROUTING || ismType === IsmType.FALLBACK_ROUTING) {
+      // The current logic performs does some checks to see if the current signer is the owner.
+      // If not owner, it will under the hood create new ISMs if necessary. In the future, we can optimise
+      // this so that it returns a minimum set of transactions to update the ISM in-place.
+      contract = await this.updateRoutingIsm({
+        destination,
+        config,
+        origin,
+        existingIsmAddress: this.args.addresses.deployedIsm,
+        logger,
+      });
+    } else {
+      contract = await EvmIsmModule.deploy({
+        destination,
+        config,
+        multiProvider: this.multiProvider,
+        factories: this.args.addresses,
+        deployer: this.deployer,
+      });
+    }
+
+    // if update was in-place, there's no change in address
+    this.args.addresses.deployedIsm = contract.address;
+    this.args.config = config;
     return [];
   }
 
   // manually write static create function
-  public static async create({
-    chain,
-    config,
-    deployer,
-    factories,
-    multiProvider,
-  }: {
+  public static async create(params: {
     chain: ChainNameOrId;
     config: IsmConfig;
     deployer: HyperlaneDeployer<any, any>;
-    factories: HyperlaneContracts<ProxyFactoryFactories>;
+    factories: HyperlaneAddresses<ProxyFactoryFactories>;
     multiProvider: MultiProvider;
   }): Promise<EvmIsmModule> {
+    const { chain, config, deployer, factories, multiProvider } = params;
     const destination = multiProvider.getChainName(chain);
-    const ismCreator = new EvmIsmCreator(deployer, multiProvider, factories);
-    const deployedIsm = await ismCreator.deploy({
+    const deployedIsm = await EvmIsmModule.deploy({
```

Comment:
why have a `deploy` separate from `create`?

Reply:
the `deploy` is an implementation detail of `create` that can get reused in `update` - would keep this split here to distinguish between returning a full module and returning just an individual deployed ism (which may be top-level or a submodule of some other ism)